clouddriver-core/src/main/groovy/com/netflix/spinnaker/clouddriver/cache/LoggingInstrumentation.groovy

```diff
@@ -33,7 +33,8 @@ class LoggingInstrumentation implements ExecutionInstrumentation {
 
   @Override
   void executionCompleted(Agent agent) {
-    logger.info("${agent.providerName}:${agent.agentType} completed")
+    def durationMs = Math.round((System.nanoTime() - MetricInstrumentation.getAgentStartTimeNs(agent)) / 1000000)
```

Comment:
I'd be pretty ok with changing ExecutionInstrumentation callback to get passed the elapsed time and have the AgentScheduler runnable deal with tracking the elapsed time and killing off that whole dirty threadlocal